activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -40,7 +40,7 @@ class Mysql2Adapter < AbstractMysqlAdapter
 
       def initialize(connection, logger, connection_options, config)
         super
-        @prepared_statements = false unless config.key?(:prepared_statements)
+        self.prepared_statements = false unless config.key?(:prepared_statements)
```

Comment:
This isn't quite right. It will set the value for the current thread, but the default value will already have been set in the `super` call. So other threads won't necessarily behave correctly.

The simplest fix is just to overwrite the instance variable here with a new `ThreadLocalVar` that has the correct default. The alternative would be to communicate the default to the superclass implementation, but that seems like a pain.

Reply:
Ah, good catch. Something like `super(connection, logger, connection_options, config.reverse_merge(prepared_statements: false))` doesn't seem too terrible, but I'm happy with overwriting the instance variable too. 👍🏻

Reply:
Oh, well if we're allowed to pass a modified config, then I like your idea better.

Reply:
Heh, once I change this and fix the mysql behavior, the test I added fails on mysql, because the test assumes that prepared statements are enabled by default. So I'm changing the test to be conditional.